test/tests/arch.bats

```diff
@@ -36,3 +36,68 @@ load ../util/util
     [[ "${status}" == "12" ]]
     [[ "${output}" == "[!] testpkg is not available for the 'amd64' architecture." ]]
 }
+
+@test "allow arch all in sub packages" {
+    package_testpkg-doc() {
+      arch=('all')
+    }
+
+    package_testpkg() {
+      echo ''
+    }
+
+    pkgbuild string maintainer1 'Foo Bar <foo@bar.com>'
+    pkgbuild array pkgname testpkg testpkg-doc
+    pkgbuild string pkgver 1.0.0
+    pkgbuild string pkgrel 1
+    pkgbuild string pkgdesc "package description"
+    pkgbuild array arch amd64
+    pkgbuild function package_testpkg
+    pkgbuild function package_testpkg-doc
+    pkgbuild clean
+    makedeb --lint
+}
+
+@test "allow arch all in sub packages - check architecture field" {
+    package_testpkg-doc() {
+      arch=('all')
+    }
+
+    package_testpkg() {
+      echo ''
+    }
+
+    pkgbuild string maintainer1 'Foo Bar <foo@bar.com>'
+    pkgbuild array pkgname testpkg testpkg-doc
+    pkgbuild string pkgver 1.0.0
+    pkgbuild string pkgrel 1
+    pkgbuild string pkgdesc "package description"
+    pkgbuild array arch amd64
+    pkgbuild function package_testpkg
+    pkgbuild function package_testpkg-doc
+    pkgbuild clean
+    makedeb -d
+    [[ "$(cat pkg/testpkg-doc/DEBIAN/control | grep '^Architecture')" == "Architecture: all" ]]
+}
```

Comment:
You can just remove the previous test in favor of this one, right? The stuff that `makedeb --lint` performs happens during normal builds as well, so if that's the only difference you should only need this one.

Reply:
Done.